.github/workflows/test-query-engine.yml

```diff
@@ -137,23 +159,36 @@ jobs:
       relation_load_strategy: '["query"]'
 
   mssql:
-    if: github.event_name == 'push' || (github.event_name == 'pull_request' && matrix.run_on_pr)
     strategy:
       fail-fast: false
       matrix:
         database:
           - name: 'mssql_2022'
             version: '2022'
-            run_on_pr: true
+            ubuntu: '24.04'
+    name: ${{ matrix.database.name }}
+    uses: ./.github/workflows/test-query-engine-template.yml
+    with:
+      name: ${{ matrix.database.name }}
+      version: ${{ matrix.database.version }}
+      ubuntu: ${{ matrix.database.ubuntu }}
+      connector: 'sqlserver'
+      relation_load_strategy: '["query"]'
+
+  mssql-push:
+    if: github.event_name == 'push'
+    strategy:
+      fail-fast: false
+      matrix:
+        database:
           - name: 'mssql_2019'
             version: '2019'
-            run_on_pr: false
+            ubuntu: '24.04'
           - name: 'mssql_2017'
             version: '2017'
             ubuntu: '20.04'
```

Comment:
This part should be gone already. Did we have a merge conflict that reintroduced this? Everythign shoudl run against ubuntu-latest and we do not test mssql2017 anymore.

Reply:
This was actually on `main`. Yeah, some wrong merge conflict resolution should have re-introduced this. Removed it.